### DISCUSSION TITLE
How do I remove complex values from an array of eigenvalues?

### DISCUSSION THREAD

**Discussion by amca01** [josdejong/mathjs#3121] [Q&A]
I'm using the `math.eigs` function to compute all roots of a polynomial, with the eigenvalues of the companion matrix.  However, I am only interested in real roots, and some values returned are complex.  I can see all the roots with

```
 var evs = math.eigs(A, {eigenvectors: false, precision: 1e-6}).values;
 console.log(math.typeOf(evs));
 console.log(math.count(evs));
 console.log(math.size(evs));
 console.log(math.format(evs,6));
```
but I can't find out how to isolate individual values, or to remove the complex values.  This, for example:

```
console.log(math.format(evs[0],6));
```
returns an error.  (So does `math.size` above.)  So I have two questions: (1) how do I access individual eigenvalues, and (2) is there a built-in way of determining if a number is real or complex?  (In my case, I could simply ask if the imaginary part of a value was less than the precision, of course.)  In any case, I need to remove the complex values, leaving only reals.

Thank you!

**Comment by josdejong** [comment]
The returned `values` is either an Array or a Matrix (depending on whether the input was an Array or Matrix). Getting a value out of an Array goes like `values[0]`, and if the result is a Matrix you can do `values.get([0])`. You can look up the docs for more information on how to work with arrays and matrices.

To filter out complex values you can use functions like `math.filter` and `math.isComplex`.
